appservice/src/connectToGitHub.ts

```diff
@@ -14,76 +13,75 @@ import { ext } from './extensionVariables';
-    requestOptions.url = 'https://api.github.com/user/orgs';
-    const gitHubOrgs: Object[] = await getJsonRequest(requestOptions, node);
-    const orgQuickPicks: IAzureQuickPickItem<{}>[] = createQuickPickFromJsons([gitHubUser], 'login', undefined, ['repos_url']).concat(createQuickPickFromJsons(gitHubOrgs, 'login', undefined, ['repos_url']));
-    const orgQuickPick: gitHubOrgData = (await ext.ui.showQuickPick(orgQuickPicks, { placeHolder: 'Choose your organization.' })).data;
-    let repoQuickPick: gitHubReposData;
-    requestOptions.url = nonNullProp(orgQuickPick, 'repos_url');
-    const gitHubRepos: Object[] = await getJsonRequest(requestOptions, node);
-    let repoQuickPicks: IAzureQuickPickItem<{}>[] = createQuickPickFromJsons(gitHubRepos, 'name', undefined, ['url', 'html_url']);
-    do {
-        if (requestOptions.nextLink && requestOptions.url !== requestOptions.lastLink) {
-            // this makes sure that a nextLink exists and that the last requested url wasn't the last page
-            repoQuickPicks.push({
-                label: '$(sync) Load More',
-                description: '',
-                data: {
-                    url: requestOptions.nextLink
-                },
-                suppressPersistence: true
-            });
-        }
-        repoQuickPick = (await ext.ui.showQuickPick(repoQuickPicks, { placeHolder: 'Choose project.' })).data;
-
-        if (repoQuickPick.url === requestOptions.nextLink) {
+    try {
+        await signRequest(requestOptions, new TokenCredentials(oAuth2Token));
+        requestOptions.url = 'https://api.github.com/user';
+        const gitHubUser: Object[] = await getJsonRequest(requestOptions);
+        requestOptions.url = 'https://api.github.com/user/orgs';
+        const gitHubOrgs: Object[] = await getJsonRequest(requestOptions);
+        const orgQuickPicks: IAzureQuickPickItem<{}>[] = createQuickPickFromJsons([gitHubUser], 'login', undefined, ['repos_url']).concat(createQuickPickFromJsons(gitHubOrgs, 'login', undefined, ['repos_url']));
+        const orgQuickPick: gitHubOrgData = (await ext.ui.showQuickPick(orgQuickPicks, { placeHolder: 'Choose your organization.' })).data;
+        let repoQuickPick: gitHubReposData;
+        requestOptions.url = nonNullProp(orgQuickPick, 'repos_url');
+        const gitHubRepos: Object[] = await getJsonRequest(requestOptions);
+        let repoQuickPicks: IAzureQuickPickItem<{}>[] = createQuickPickFromJsons(gitHubRepos, 'name', undefined, ['url', 'html_url']);
+        while (requestOptions.nextLink) {
+            // load all the next repos at once
             requestOptions.url = nonNullProp(requestOptions, 'nextLink');
-            // remove the stale Load More quick pick
-            repoQuickPicks.pop();
-            const moreGitHubRepos: Object[] = await getJsonRequest(requestOptions, node);
+            const moreGitHubRepos: Object[] = await getJsonRequest(requestOptions);
```

Comment:
You can't just blindly get all repos because we have no idea how long that'll take. I suggest you add some sort of reasonable timeout and still have a 'load more' if that timeout is exceeded